api/src/main/java/com/velocitypowered/api/newcommand/Command.java

```diff
@@ -0,0 +1,67 @@
+package com.velocitypowered.api.newcommand;
+
+import com.velocitypowered.api.command.CommandSource;
+import com.velocitypowered.api.proxy.Player;
+
+/**
+ * Represents a command that can be executed by a {@link CommandSource},
+ * such as a {@link Player} or the console.
+ *
+ * @param <C> the type of the execution context
+ */
+public interface Command<C extends CommandExecutionContext> {
+
+    /**
+     * Represents the command type, which determines
+     * its parsing capabilities.
+     */
+    // TODO Introduce CommandMeta and move there?
```

Comment:
I think this is a good idea

Reply:
I'm leaning more towards `Command` containing the immutable meta and have an associated `CommandExecutor`. This way we can have builders that automatically register commands on construction (similar to the scheduler system). Also, we can get rid of the `Type` enum to let users provide their own `Command` implementations (i.e. get rid of `RawCommand`).

Reply:
I'm happy to run with this idea (`Command.Type` is kind of a kludge anyway).